### PR TITLE
refactor:TOC and sticky

### PR DESCRIPTION
**Pull request by JeffreyMa597**

### CODE REVIEW

File: src/components/BaseSeparatePage/ContributorsDialog/index.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import styles from './index.module.scss'
 type ContributorsDialogType = {
   contributors: Array<Author>
   status: boolean
+  author?: Author
```
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/ContributorsDialog/index.tsx:13]: Does not appear to be used.

File: src/components/BaseSeparatePage/TableOfContents/index.tsx [hunk 2]
```diff
@@ -1,50 +1,115 @@
...
+  isInViewport: boolean
+}
+
+export interface TOCContextValue {
+  tocItems: TOCItemInfo[]
+  addTOCItem: (tocItemEl: HTMLElement, title: string) => void
+  removeTOCItem: (tocItemEl: HTMLElement) => void
+}
+
+export const TOCContext = createContext<TOCContextValue>(
```
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/TableOfContents/index.tsx:27]: TOCContext and TOCItem should be elevated to a public component that provides the base TOC logic and then internally wraps a new StyledTOCList based on it wherever needed (e.g. in BaseSeparatePage) .

File: src/components/BaseSeparatePage/TableOfContents/index.tsx
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/TableOfContents/index.tsx:1]: useIntersectionObserver and useHeadingsData files need remove.

File: src/components/BaseSeparatePage/index.tsx [hunk 3]
```diff
@@ -84,24 +83,25 @@ export const BaseSeparatePage: FC<BaseSeparatePageType> = props => {
   }) => (
     <div className={styles.functionsWrap}>
       <Functions isProgressBar={isProgressBar} functions={functions} className={className} />
-      {/* Todo: progressbar need complete later*/}
-      {isProgressBar ? <TableOfContents className={clsx(styles.tableOfContents)} editLink={editLink} /> : null}
+      {/* {isProgressBar ? <TableOfContents className={clsx(styles.tableOfContents)} editLink={editLink} /> : null} */}
```
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/index.tsx:86]: Is it written wrong here?

File: src/components/BaseSeparatePage/index.tsx [hunk 4]
```diff
@@ -84,24 +83,25 @@ export const BaseSeparatePage: FC<BaseSeparatePageType> = props => {
...
     </div>
   )
 
   const [isOpen, setIsOpen] = useState<boolean>(false)
   const handleContributorDialogClose = () => setIsOpen(false)
 
+  const TOCContextConsumer = TOCContext.Consumer
```
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/index.tsx:93]: Does not appear to be used.

**Review by WhiteMinds — CHANGES_REQUESTED** [review]

File: src/components/BaseSeparatePage/ContributorsDialog/index.tsx (hunk 1 above)
Comment (JeffreyMa597) [reply src/components/BaseSeparatePage/ContributorsDialog/index.tsx:13]: removed.

File: src/components/BaseSeparatePage/TableOfContents/index.tsx
Comment (JeffreyMa597) [reply src/components/BaseSeparatePage/TableOfContents/index.tsx:1]: removed.

File: src/components/BaseSeparatePage/index.tsx (hunk 3 above)
Comment (JeffreyMa597) [reply src/components/BaseSeparatePage/index.tsx:87]: in our original logic, the `FunctionsContainer ` need to be compatible with and without a TOC.

Comment (WhiteMinds) [reply src/components/BaseSeparatePage/index.tsx:87]: I mean it doesn't look like it should be annotated out here?
If it is annotated, there will be no opportunity to show the TOC.

**Pull request closed without merging**